### PR TITLE
Add documentation for Docker.

### PR DESCRIPTION
**Pull request by david-antliff-imgtec**
Signed-off-by: David Antliff <david.antliff@imgtec.com>

### CODE REVIEW

**Comment by david-antliff-imgtec** [comment]
retest this please

**Comment by david-antliff-imgtec** [comment]
retest this please

File: doc/docker.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+See the [Docker documentation](https://docs.docker.com) for more information about Docker.
+
+## Building the docker image
+
+First, install Docker as per the [Docker documentation](https://docs.docker.com/engine/installation/). This guide assumes that the Docker installation process has created a *docker* user, avoiding the need to run `sudo` with docker commands.
+
+Clone the Awa LWM2M repository from [GitHub](https://github.com/FlowM2M/AwaLWM2M):
+
+``` $ git clone https://github.com/FlowM2M/AwaLWM2M.git```
```
Comment (seank-img) [inline doc/docker.md]: add note about using an existing Awa repo - ```make clean``` works fine.

File: doc/docker.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Note: when Awa builds, it runs a suite of unit tests to check that it is functioning correctly. Many of these tests intentionally trigger errors, which will be displayed in the docker build log as red errors. These can be ignored.
+
+## Using the container
+
+Once you have a docker image, you may now run Awa binaries directly.
+
+To run the bootstrap daemon (with the localhost bootstrap config):
+
+``` $ docker run --rm -p 15685:15685/udp --name awa_bootstrapd -it flowm2m/awalwm2m awa_bootstrapd --config core/bootstrap-docker.config```
```
Comment (seank-img) [inline doc/docker.md]: This didn't work, change to...

```docker run --rm -p 6000:6000/udp -p 12345:12345/udp --link awa_bootstrapd:awa_bootstrapd --link awa_serverd:awa_serverd --name awa_clientd -it flowm2m/awalwm2m awa_clientd --endPointName Client1 --bootstrap coap://awa_bootstrapd:15685```

File: doc/docker.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+To run the server daemon:
+
+``` $ docker run --rm -p 5683:5683/udp -p 54321:54321/udp --name awa_serverd -it flowm2m/awalwm2m awa_serverd```
+
+To run the client daemon, which will bootstrap and register with the server:
+
+``` $ docker run --rm -p -p 6000:6000/udp 12345:12345/udp --link awa_bootstrapd:awa_bootstrapd --link awa_serverd:awa_serverd --name awa_clientd -it flowm2m/awalwm2m awa_clientd --endpointName Client1 --bootstrap coap://awa_bootstrapd:5683```
```
Comment (seank-img) [inline doc/docker.md]: This didn't work, change to...

```docker run --rm -p 6000:6000/udp -p 12345:12345/udp --link awa_bootstrapd:awa_bootstrapd --link awa_serverd:awa_serverd --name awa_clientd -it flowm2m/awalwm2m awa_clientd --endPointName Client1 --bootstrap coap://awa_bootstrapd:15685```

File: doc/docker.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+Awa tools may also be used:
+
+``` $ docker run --rm --link awa_serverd:ipc -it flowm2m/awalwm2m awa-server-list-clients -a ipc```
+
+``` $ docker run --rm --link awa_clientd:ipc -it flowm2m/awalwm2m awa-client-get -a ipc /3```
+
+``` $ docker run --rm --link awa_clientd:ipc -it flowm2m/awalwm2m awa-server-read -a ipc --clientID Client1 /3/0```
```
Comment (seank-img) [inline doc/docker.md]: Change this to...

```docker run --rm --link awa_serverd:ipc -it flowm2m/awalw2m2 awa-server-read -a ipc --clientID Client1 /3/0```

File: doc/docker.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+
+## Ease of use
+
+Obviously having to specify all those docker parameters each time is annoying. To make this easier, create the following aliases:
+
+``` $ alias ac docker run --rm --link awa_clientd:ipc -it flowm2m/awalwm2m```
```
Comment (seank-img) [inline doc/docker.md]: didn't work for me, I had to use...

```alias ac="docker run --rm --link awa_clientd:ipc -it flowm2m/awalw2m2"```

File: doc/docker.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+``` $ alias as docker run --rm --link awa_serverd:ipc -it flowm2m/awalwm2m```
```
Comment (seank-img) [inline doc/docker.md]: as above...

**Comment by seank-img** [comment]
+1 when comments addressed

**Pull request merged**